apps/acrs/behaviours/resume-form-session.js

```diff
@@ -0,0 +1,203 @@
+
+        if (singleCase && noCaseOrSameCase && !multipleCasesInSession) {
+          if (cases[0]) {
+            this.setupSession(req, cases[0].session);
+          }
+          req.sessionModel.set('multipleCases', false);
+          return res.redirect(super.getNextStep(req, res, next));
+        }
+
+        const filteredCases = this.addCasesToSession(req, cases, idNumber);
+        this.setupRadioButtons(req, filteredCases.filter(form => form.id));
+
+        return super.configure(req, res, next);
+      })
+      .catch(next);
+  }
+
+  addCasesToSession(req, cases, idNumber) {
+    const idType = req.sessionModel.get('id-type');
+    let caseAlreadyInSession = '';
+
+    if (idType === 'brp') {
+      caseAlreadyInSession = cases.find(obj => obj.session.brp === idNumber);
+    }else{
+      caseAlreadyInSession = cases.find(obj => obj.session.uan === idNumber);
+    }
+
+    if (!caseAlreadyInSession) {
+      this.addSessionCaseToList(req, cases);
+    }
+
+    const filteredCases = cases.filter(caseType => {
+      const isDuplicate = cases.filter(obj => {
+        if (idType === 'brp') {
+          return obj.session.brp === caseType.session.brp;
+        }
+        if (idType === 'uan') {
+          return obj.session.uan === caseType.session.uan;
+        }
+      }).length > 1;
+
+      return caseType.id || (!caseType.id && !isDuplicate);
+    });
+
+    req.sessionModel.set('user-cases', filteredCases);
+    return filteredCases;
+  }
+
+  addSessionCaseToList(req, cases) {
+    const defaultSession = {
+      session: Object.assign(...SESSION_DEFAULTS.fields.map(field => {
+        return { [field]: req.sessionModel.get(field) };
+      }))
+    };
+    defaultSession.session.steps = SESSION_DEFAULTS.steps;
+    cases.push(defaultSession);
+  }
+
+  parseCasesSessions(body) {
+    let cases = body;
+
+    if (cases.length) {
+      cases = cases.map(caseType => {
+        const session = caseType.session;
+        caseType.session = typeof session === 'string' ?
+          JSON.parse(session) : session;
+
+        return caseType;
+      });
+    }
+    return cases;
+  }
+  // POST lifecycle
+  saveValues(req, res, next) {
+    const cases = req.sessionModel.get('user-cases') || [];
+    const idType = req.sessionModel.get('id-type');
+    let caseObj = [];
+
+    if (idType === 'brp') {
+      caseObj = cases.find(obj => obj.session.brp === req.form.values['id-type']);
+    }else{
+      caseObj = cases.find(obj => obj.session.uan === req.form.values['id-type']);
+    }
+
+    if (caseObj) {
+      req.sessionModel.set('id', caseObj.id);
+    }
+    this.setupSession(req, caseObj.session);
+
+    req.sessionModel.set('multipleCases', true);
+
+    return super.saveValues(req, res, next);
+  }
+
+  setupRadioButtons(req, cases) {
+    let dob = '';
+    let idNumber = '';
+    let idLabel = '';
+    const idType = req.sessionModel.get('id-type');
+
+    req.form.options.fields['id-type'] = {
```

Comment:
Can I suggest a field name that isn't already the name of a piece of session data?
Maybe 'referral'?

Reply:
referral sounds let an object/person.

Reply:
Ok, but this is the name of the radio group which to me is a series of referrals to choose from. You can call it what you like, but I think id-type is too confusing.